app/src/main/java/com/findmyrecycling/MainActivity.kt

```diff
@@ -29,84 +29,92 @@ class MainActivity : ComponentActivity() {
-            modifier = Modifier
-                .fillMaxWidth()
-                .padding(start = 4.dp, end = 4.dp, top = 2.dp, bottom = 2.dp)
-                .border(width = 1.dp,
+
+
+    @Composable
+    fun RecycleSearch(product: String) {
+        var recyclable by remember { mutableStateOf("") }
+        var location by remember { mutableStateOf("") }
+
+        Column {
+            OutlinedTextField(
+                value = recyclable,
+                onValueChange = { recyclable = it },
+                label = { Text(stringResource(R.string.recyclable)) },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 4.dp, end = 4.dp, top = 2.dp, bottom = 2.dp)
+                    .border(
+                        width = 1.dp,
                         color = Color.Black,
-                        shape = RoundedCornerShape(4.dp)),
-        )
-
-        OutlinedTextField(
-            value = location,
-            onValueChange = { location = it },
-            label = { Text(stringResource(R.string.location)) },
-            modifier = Modifier
-                .fillMaxWidth()
-                .padding(start = 4.dp, end = 4.dp, top = 2.dp, bottom = 2.dp)
-                .border(width = 1.dp,
+                        shape = RoundedCornerShape(4.dp)
+                    ),
+            )
+
+            OutlinedTextField(
+                value = location,
+                onValueChange = { location = it },
+                label = { Text(stringResource(R.string.location)) },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 4.dp, end = 4.dp, top = 2.dp, bottom = 2.dp)
+                    .border(
+                        width = 1.dp,
                         color = Color.Black,
-                        shape = RoundedCornerShape(4.dp)),
-        )
+                        shape = RoundedCornerShape(4.dp)
+                    ),
+            )
 
 
+        }
     }
-}
 
-@Preview(showBackground = true)
-@Composable
-fun DefaultPreview() {
-    FindMyRecyclingTheme {
-        RecycleSearch("Android")
-    }
+        @Preview(showBackground = true)
+        @Composable
+        fun DefaultPreview() {
+            FindMyRecyclingTheme {
+                RecycleSearch("Android")
+            }
 
-    Button(
-        onClick = {
-            signIn()
+            Button(
+                onClick = {
+                    signIn()
+                }
+            ) {
+                Text(text = "Logon")
+            }
+        }
+
+        fun signIn() {
+            val providers = arrayListOf(AuthUI.IdpConfig.EmailBuilder().build())
+            val signInIntent = AuthUI.getInstance()
+                .createSignInIntentBuilder()
+                .setAvailableProviders(providers)
+                .build()
+
+            signInLauncher.launch(signInIntent)
+        }
+
```

Comment:
From line 97- 105 is where the edit for this file change is, there was a typo on line 99 with the signInIntent which aslo gets called on line 104, reference for this code change comes from best practices found here: 
https://kotlinlang.org/docs/coding-conventions.html